### ISSUE TITLE
brew fails to install avog2s

### ISSUE THREAD

**Issue by alexjcwitsil** [ibrewster/homebrew-avo#9]
ran `brew install avog2s` and got the following output (attached).
[avog2s_error_log.txt](https://github.com/ibrewster/homebrew-avo/files/9087052/avog2s_error_log.txt)
With these warnings and errors:
Warning: Building gcc from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building m4 from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building openssl@1.1 from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building gdbm from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building perl from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building pkg-config from source:
  The bottle needs to be installed into /home/linuxbrew/.linuxbrew.

```
Warning: Building ncurses from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building gettext from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building libidn2 from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building ruby from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building util-linux from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building openldap from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building krb5 from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building eccodes from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Warning: Building open-mpi from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/ajwitsil/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/ajwitsil/.linuxbrew)
Error: avog2s: SHA256 mismatch
Expected: 6ed2130eed7b741df3b19052b29b3324601403581c7b9afb015e0370e299a2bd
  Actual: 5b906e79a66bce5fbf8ac1e9d55fcc3366a86cb42762f910331a06794596fbf2
    File: /home/ajwitsil/.cache/Homebrew/downloads/258a8fae83c0b8a362e27579c115a9251c4bdd9738846e7c930f2a8bc73ae2ac--SHTOOLS-4.7.1.tar.gz
To retry an incomplete download, remove the file above.
```

 I tried removing the file: /home/ajwitsil/.cache/Homebrew/downloads/258a8fae83c0b8a362e27579c115a9251c4bdd9738846e7c930f2a8bc73ae2ac--SHTOOLS-4.7.1.tar.gz
 as was suggested and then reinstalling, but the install failed again.  
 Any ideas on how I could install avog2s? 
 Also, I should note I am trying to install on a remote server, on which I don't have sudo privileges.

**Comment by ibrewster** [comment]
In my testing, I'm seeing two issues:

1) The one you just reported, which I have fixed by replacing the requires line in the formula (pull the new version and you should get past that)
2) The formula relies on downloading some code from https://agupubs.onlinelibrary.wiley.com. Unfortunately, it looks like they have added a layer of DDoS security to their system that is breaking the homebrew curl command. I'm not sure immediately how to work around this issue

**Comment by jwbishop** [comment]
I suspect that the code from WILEY is the Horizontal Wind Model (HWM14). My installed copy is 664 kB. Would it be worth having a local version of this code in the repo for stability?

**Comment by ibrewster** [comment]
Yes, it is the HWM14 code (sorry for not specifying that!). Assuming the license allows us to, yes having a local version of the code in the repo would be an easy solution to the problem.

**Comment by liamtoney** [comment]
My only comment from scanning this is to note that @alexjcwitsil is using linuxbrew (Linux), not homebrew (macOS). Might not be relevant.

**Comment by ibrewster** [comment]
I was able to reproduce the issue on my mac without difficulty, so yeah, I don't think it makes a difference :)

**Comment by ibrewster** [comment]
Ok, I pulled the HWM14 code into the repository, and updated the formula to point to that "local" copy. This should fix the avog2s formula, but unfortunately I'm now having difficulty with the volcano-ash3d-metreader formula, on which avog2s depends. If you can get that one to install (it MAY be an OS issue), then you should be good...

**Comment by ibrewster** [comment]
It looks like the issue was that the homebrew include/lib dirs had changed on my mac since I first made the makefiles (probably a homebrew version difference). I have updated the relevant formulas to look for the new homebrew paths, and can now successfully install avog2s on my mac. I suspect linuxbrew would have worked properly since I included the hwm14 code in the repository, as I was already looking for the linuxbrew paths. Closing this issue as resolved, but let me know if you continue to have issues.
